<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>991</width>
    <height>451</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>991</width>
    <height>451</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>991</width>
    <height>451</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Basic3D</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="GLWidget" name="openGLWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>431</width>
      <height>431</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>10</y>
      <width>82</width>
      <height>431</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="mTranslate">
     <item>
      <layout class="QHBoxLayout" name="mTSliders">
       <item>
        <widget class="QSlider" name="sliderMx">
         <property name="minimum">
          <number>-500</number>
         </property>
         <property name="maximum">
          <number>500</number>
         </property>
         <property name="pageStep">
          <number>1</number>
         </property>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="tickPosition">
          <enum>QSlider::NoTicks</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="sliderMy">
         <property name="minimum">
          <number>-500</number>
         </property>
         <property name="maximum">
          <number>500</number>
         </property>
         <property name="pageStep">
          <number>1</number>
         </property>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="sliderMz">
         <property name="minimum">
          <number>-500</number>
         </property>
         <property name="maximum">
          <number>500</number>
         </property>
         <property name="pageStep">
          <number>1</number>
         </property>
         <property name="value">
          <number>0</number>
         </property>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="mTLabels">
       <item>
        <widget class="QLabel" name="labelMx">
         <property name="text">
          <string>Mx</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="labelMy">
         <property name="text">
          <string>My</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="labelMz">
         <property name="text">
          <string>Mz</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>10</y>
      <width>63</width>
      <height>431</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="rotationLayout">
     <item>
      <layout class="QHBoxLayout" name="mRotate">
       <item>
        <layout class="QVBoxLayout" name="mRButtons">
         <item>
          <widget class="QRadioButton" name="radioX">
           <property name="text">
            <string>X</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radioY">
           <property name="text">
            <string>Y</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radioZ">
           <property name="text">
            <string>Z</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QSlider" name="sliderRotate">
         <property name="minimum">
          <number>-360</number>
         </property>
         <property name="maximum">
          <number>360</number>
         </property>
         <property name="pageStep">
          <number>1</number>
         </property>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QLabel" name="rotationLabel">
       <property name="text">
        <string>Rotation</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>10</y>
      <width>56</width>
      <height>431</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="zoomLayout">
     <item>
      <layout class="QHBoxLayout" name="zoomControls">
       <item>
        <widget class="QSlider" name="sliderZoom">
         <property name="maximum">
          <number>200</number>
         </property>
         <property name="pageStep">
          <number>10</number>
         </property>
         <property name="value">
          <number>100</number>
         </property>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QVBoxLayout" name="zoomLabels">
         <item>
          <widget class="QLabel" name="maxZoom">
           <property name="text">
            <string>2X</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="minZoom">
           <property name="text">
            <string>0.1X</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QLabel" name="zoomLabel">
       <property name="text">
        <string>Zoom</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget_3">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>10</y>
      <width>82</width>
      <height>431</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="cTranslate">
     <item>
      <layout class="QHBoxLayout" name="cTSliders">
       <item>
        <widget class="QSlider" name="sliderCx">
         <property name="minimum">
          <number>-500</number>
         </property>
         <property name="maximum">
          <number>500</number>
         </property>
         <property name="pageStep">
          <number>1</number>
         </property>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="tickPosition">
          <enum>QSlider::NoTicks</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="sliderCy">
         <property name="minimum">
          <number>-500</number>
         </property>
         <property name="maximum">
          <number>500</number>
         </property>
         <property name="pageStep">
          <number>1</number>
         </property>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="sliderCz">
         <property name="minimum">
          <number>-500</number>
         </property>
         <property name="maximum">
          <number>500</number>
         </property>
         <property name="pageStep">
          <number>1</number>
         </property>
         <property name="value">
          <number>0</number>
         </property>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="cTLabels">
       <item>
        <widget class="QLabel" name="labelCx">
         <property name="text">
          <string>Cx</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="labelCy">
         <property name="text">
          <string>Cy</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="labelCz">
         <property name="text">
          <string>Cz</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>11</y>
      <width>101</width>
      <height>435</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <layout class="QVBoxLayout" name="clippingLayout">
       <item>
        <layout class="QHBoxLayout" name="nearLayout">
         <item>
          <widget class="QLabel" name="labelNear">
           <property name="text">
            <string>Near:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="nearSpinBox">
           <property name="decimals">
            <number>1</number>
           </property>
           <property name="minimum">
            <double>1.000000000000000</double>
           </property>
           <property name="maximum">
            <double>1000.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>1.000000000000000</double>
           </property>
           <property name="value">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="farLayout">
         <item>
          <widget class="QLabel" name="labelFar">
           <property name="text">
            <string>Far:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="farSpinBox">
           <property name="decimals">
            <number>1</number>
           </property>
           <property name="minimum">
            <double>1.000000000000000</double>
           </property>
           <property name="maximum">
            <double>1000.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.000000000000000</double>
           </property>
           <property name="value">
            <double>1000.000000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="mouseLayout">
       <item>
        <widget class="QRadioButton" name="mouseModel">
         <property name="text">
          <string>Model</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="mouseCamera">
         <property name="text">
          <string>Camera</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QPushButton" name="fillModeToggle">
       <property name="text">
        <string>Toggle Fill</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="resetButton">
       <property name="text">
        <string>Reset</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="openFileButton">
       <property name="text">
        <string>Open File</string>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="vCTLayout">
       <item>
        <layout class="QVBoxLayout" name="vCTLabels">
         <item>
          <widget class="QLabel" name="vCxLabel">
           <property name="text">
            <string>Cx:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="vCyLabel">
           <property name="text">
            <string>Cy:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="vCzLabel">
           <property name="text">
            <string>Cz:</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="editCTLabels">
         <item>
          <widget class="QLineEdit" name="editCx">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>0</string>
           </property>
           <property name="maxLength">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="editCy">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>0</string>
           </property>
           <property name="maxLength">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="editCz">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>0</string>
           </property>
           <property name="maxLength">
            <number>4</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="vTLayout">
       <item>
        <layout class="QVBoxLayout" name="vTLabels">
         <item>
          <widget class="QLabel" name="vMxLabel">
           <property name="text">
            <string>Mx:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="vMyLabel">
           <property name="text">
            <string>My:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="vMzLabel">
           <property name="text">
            <string>Mz:</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="editTLabels">
         <item>
          <widget class="QLineEdit" name="editMx">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>0</string>
           </property>
           <property name="maxLength">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="editMy">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>0</string>
           </property>
           <property name="maxLength">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="editMz">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>0</string>
           </property>
           <property name="maxLength">
            <number>4</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="vRLayout">
       <item>
        <layout class="QVBoxLayout" name="vRLabels">
         <item>
          <widget class="QLabel" name="vRxLabel">
           <property name="text">
            <string>Rx:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="vRyLabel">
           <property name="text">
            <string>Ry:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="vRzLabel">
           <property name="text">
            <string>Rz:</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="editRLabels">
         <item>
          <widget class="QLineEdit" name="editRx">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>0</string>
           </property>
           <property name="maxLength">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="editRy">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>0</string>
           </property>
           <property name="maxLength">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="editRz">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>0</string>
           </property>
           <property name="maxLength">
            <number>4</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>920</x>
      <y>10</y>
      <width>62</width>
      <height>80</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="colorLayout">
     <item>
      <layout class="QHBoxLayout" name="layoutR">
       <item>
        <widget class="QLabel" name="labelR">
         <property name="text">
          <string>R:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="spinBoxR">
         <property name="decimals">
          <number>1</number>
         </property>
         <property name="minimum">
          <double>0.000000000000000</double>
         </property>
         <property name="maximum">
          <double>1.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
         <property name="value">
          <double>1.000000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="layoutG">
       <item>
        <widget class="QLabel" name="labelG">
         <property name="text">
          <string>G:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="spinBoxG">
         <property name="decimals">
          <number>1</number>
         </property>
         <property name="minimum">
          <double>0.000000000000000</double>
         </property>
         <property name="maximum">
          <double>1.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
         <property name="value">
          <double>1.000000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="layoutB">
       <item>
        <widget class="QLabel" name="labelB">
         <property name="text">
          <string>B:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="spinBoxB">
         <property name="decimals">
          <number>1</number>
         </property>
         <property name="minimum">
          <double>0.000000000000000</double>
         </property>
         <property name="maximum">
          <double>1.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
         <property name="value">
          <double>1.000000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>920</x>
      <y>100</y>
      <width>61</width>
      <height>96</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="fieldOfView">
     <item>
      <layout class="QVBoxLayout" name="fovV">
       <item>
        <widget class="QLabel" name="fovLR">
         <property name="text">
          <string>FOV LR:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="vLength">
         <property name="decimals">
          <number>1</number>
         </property>
         <property name="minimum">
          <double>0.100000000000000</double>
         </property>
         <property name="maximum">
          <double>5.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
         <property name="value">
          <double>0.500000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="fovH">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>FOV TB:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="hLength">
         <property name="decimals">
          <number>1</number>
         </property>
         <property name="minimum">
          <double>0.100000000000000</double>
         </property>
         <property name="maximum">
          <double>5.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
         <property name="value">
          <double>0.500000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>glwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
